.github/workflows/ci.yml

```diff
@@ -0,0 +1,290 @@
+      - name: ci/lint
+        run: |
+          npm run check-types
+  tests:
+    runs-on: ubuntu-22.04
+    needs: [check-type, check-i18n, check-lint]
+    permissions:
+      checks: write
+      pull-requests: write
+    steps:
+      - name: ci/checkout-repo
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+      - name: ci/setup-node
+        uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # v3.6.0
+        id: setup_node
+        with:
+          node-version-file: ".nvmrc"
+          cache: npm
+      - uses: actions/cache@2b8105bdae4b746666ee225970c4a281bbd9d51f # v3.2.4
+        id: npm-cache
+        with:
+          path: |
+            '**/node_modules'
+            'e2e/playwright/node_modules'
+            'e2e/cypress/node_modules'
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}-${{ hashFiles('e2e/cypress/package-lock.json') }}-${{ hashFiles('e2e/playwright/package-lock.json') }}
+      - name: ci/get-node-modules
+        if: steps.npm-cache.outputs.cache-hit != 'true'
+        run: |
+          make node_modules
+          make e2e/playwright/node_modules
+      - name: ci/Get number of CPU cores
+        id: cpu-cores
+        uses: SimenB/github-actions-cpu-cores@c508d404ab007faae80a014072fd8c0e17792118 # v1.1.0
+      - name: ci/test
+        run: |
+          npm run test-ci -- --maxWorkers=${{ steps.cpu-cores.outputs.count }}
+      - name: Publish Test Results
+        uses: EnricoMi/publish-unit-test-result-action@a3caf02865c0604ad3dc1ecfcc5cdec9c41b7936  # v2.3.0
+        if: always()
+        with:
+          junit_files: "build/**/*.xml"
+          comment_mode: always
+          compare_to_earlier_commit: false
+  build:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: ci/checkout-repo
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+      - name: ci/setup-node
+        uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # v3.6.0
+        id: setup_node
+        with:
+          node-version-file: ".nvmrc"
+          cache: npm
+      - uses: actions/cache@2b8105bdae4b746666ee225970c4a281bbd9d51f # v3.2.4
+        id: npm-cache
+        with:
+          path: |
+            '**/node_modules'
+            'e2e/playwright/node_modules'
+            'e2e/cypress/node_modules'
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}-${{ hashFiles('e2e/cypress/package-lock.json') }}-${{ hashFiles('e2e/playwright/package-lock.json') }}
+      - name: ci/get-node-modules
+        if: steps.npm-cache.outputs.cache-hit != 'true'
+        run: |
+          make node_modules
+          make e2e/playwright/node_modules
+      - name: ci/build
+        run: |
+          node --version
+          npm version
+          make dist
+      - name: ci/persist-artifact
+        uses: actions/upload-artifact@0b7f8abb1508181956e8e162db84b466c27e18ce # v3.1.2
```

Comment:
Should this expire at some point? I don't know if GitHub limits us on stored artifacts or anything

Reply:
There is a global expiration of 90 days . But yes it should . What do you suggest ? I think 14 days is ok

Reply:
This is the current policy. 
https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#usage-limits-and-eviction-policy

Reply:
Yeah, 14 days would probably be fine. It could theoretically be helpful to save artifacts for longer, but since we only use these for test servers, we don't even really need them after a PR is closed

Reply:
The previous link was about Cache. 
GitHub documentation concerning artifacts retention policy: https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts#about-workflow-artifacts 

_"By default, GitHub stores build logs and artifacts for 90 days, and this retention period can be customized."_

Will push a commit to retain artifacts for 14 days.

Reply:
Done with: https://github.com/mattermost/mattermost-webapp/pull/12182/commits/5bd42f916c529909e890ada0946c463c643e4b99